tests/e2e/steps/ViewerContext.ts

```diff
@@ -0,0 +1,92 @@
+import { Given, When, Then, DataTable } from '@cucumber/cucumber'
+import { expect } from '@playwright/test'
+import config from '../config'
+
+import { Ocis } from '../pageObjects/Ocis'
+import { Viewer } from '../pageObjects/Viewer'
+import { uploadFile } from '../utils/helpers'
+
+const ocis = new Ocis()
+const viewer = new Viewer()
+
+Given(
+  'the following 3D models have been uploaded:',
+  async function (filesForUpload: DataTable): Promise<void> {
+    for (const file of filesForUpload.hashes()) {
+      await uploadFile(file.filename)
+    }
+  }
+)
+
+Given(
+  'the user has logged in with username {string} and password {string}',
+  async function (user: string, password: string): Promise<void> {
+    await global.page.goto(config.baseUrlOcis)
+    await ocis.login({ username: user, password: password })
+  }
+)
+
+When(
+  'the user previews the file {string} in the 3D model viewer',
+  async function (filename: string): Promise<void> {
+    await ocis.previewFile(filename)
+  }
+)
+
+Then(
+  'the 3D model {string} should be displayed in the viewport',
+  async function (filename: string): Promise<void> {
+    // check if viewport and canvas are visible
+    await expect(global.page.locator(viewer.elements.modelViewport)).toBeVisible()
+    await expect(global.page.locator(viewer.elements.modelViewportCanvas)).toBeVisible()
+    // wait for model (and its description) to be loaded
+    await global.page.waitForSelector(viewer.elements.modelViewportDescription)
+    // check if the filename is displayed in hidden h1 title element of the viewport
+    const viewportDescription = await viewer.getViewportDescription()
+    expect(viewportDescription).toContain(filename)
```

Comment:
it is on purpose, because the full Description is "Preview for <modelname>"

Reply:
@nirajacharya2  it is on purpose, because the full Description is "Preview for [modelname]", but it's a good point, in line 57, I can make the assertion stricter.

Reply:
see comment above in reply to Niraj's question. If we want to check with `toBe`, we have to add the string "Preview for " to filename...
@amrita-shrestha @saw-jan what way do you think is nicer? what do you prefer?

Reply:
yeah i read niraj 's question reply but still not satisfied with the reason and can't connect your reason with niraj question. So asking 
> Sry, can't get the reason for not using to be ? Can't we check the whole sentence with toBe?

Reply:
The method getViewportDescription() returns the text that's in the hidden H1 title which is "Preview for <filename>". An alternative way to do the assertion is removing the string "Preview for " from the result before returning the text in that function. Would you consider that a cleaner approach?

Reply:
@amrita-shrestha also if you have any other suggestion how to assert if the model is displayed in the viewport, I'm happy to hear your ideas!

Reply:
Furthermore, `false positives` cases could happen when we include `tocontain`. Something connected to this PR, I believe
https://github.com/owncloud/ocis/pull/6987#discussion_r1287912557
Our training list includes this PR for discussion as well. Perhaps we will talk about this one in the future.

Reply:
thanks everyone, refactored the getViewportDescription() and made assertion stricter by using `toBe`